From 403431b1b801326cbca732dbcc759a719c1bb7f0 Mon Sep 17 00:00:00 2001
From: Kai Uwe Broulik <kde@privat.broulik.de>
Date: Mon, 26 Feb 2024 22:09:48 +0100
Subject: [PATCH] Notifications: Only convert absolute paths to file URLs

If the contents is just e.g. a device name, there's no point
in turning it into a URL.

BUG: 481759


(cherry picked from commit f807a2793bd91360324469c5e2f1b621a62be426)
---
 libnotificationmanager/job_p.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libnotificationmanager/job_p.cpp b/libnotificationmanager/job_p.cpp
index 3d4b0165cf2..3fb45160014 100644
--- a/libnotificationmanager/job_p.cpp
+++ b/libnotificationmanager/job_p.cpp
@@ -66,6 +66,11 @@ QUrl JobPrivate::localFileOrUrl(const QString &urlString)
 {
     QUrl url(urlString);
     if (url.scheme().isEmpty()) {
+        // Likely not a path at all (e.g. just a device name).
+        if (!url.path().startsWith(QLatin1Char('/'))) {
+            return QUrl();
+        }
+
         url = QUrl::fromLocalFile(urlString);
     }
     return url;
-- 
GitLab

